astropy/units/core.py

```diff
@@ -2658,6 +2664,13 @@ def def_unit(
     unit : `~astropy.units.NamedUnit`
         The newly-defined unit, or a matching unit that was already
         defined.
+
+    Raises
+    ------
+    ValueError
+        If ``represents`` cannot be parsed as a unit, e.g., because it is
+        a malformed string or a |Quantity| that is not a scalar.
```

Comment:
I couldn't find any `def_unit()` tests with invalid strings or a non-scalar `Quantity` (at least not in `test_units.py`).